services/libs/aster-std/src/process/vm/heap.rs

```diff
@@ -2,33 +2,29 @@
 
 use core::sync::atomic::{AtomicUsize, Ordering};
 
+use crate::prelude::*;
+use crate::process::constants::{USER_HEAP_BASE, USER_HEAP_SIZE_LIMIT};
 use crate::vm::perms::VmPerms;
 use crate::vm::vmar::Vmar;
-use crate::{
-    prelude::*,
-    vm::vmo::{VmoFlags, VmoOptions},
-};
+use crate::vm::vmo::{VmoFlags, VmoOptions};
 use align_ext::AlignExt;
 use aster_rights::{Full, Rights};
 
-pub const USER_HEAP_BASE: Vaddr = 0x0000_0000_1000_0000;
-pub const USER_HEAP_SIZE_LIMIT: usize = PAGE_SIZE * 1000;
-
 #[derive(Debug)]
-pub struct UserHeap {
+pub struct Heap {
     /// the low address of user heap
-    heap_base: Vaddr,
+    bottom: Vaddr,
```

Comment:
It is more common to describe a memory range with "base" and "limit" (or "size"). Depending on how you draw the picture of a virtual address space vertically, "bottom" could either mean "base" or "base + limit". So is "top". So we should avoid using "bottom" and "top".

Even for stack, using the terms "bottom" and "top" seems to be a bad idea. Because the terms could be interpreted in one of two ways:
1. If you draw an abstract stack, the diagram is probably depicted with newer items on the top of the stack and the older ones in the bottom.
2. If you draw the memory region of a stack on x86, with higher addresses on the top and lower ones in the bottom, then newer items appear in the bottom of the stack. But on other CPU architectures, the top and bottom of the stack may be reversed.

For the reasons above, if you really want to use the terms "bottom" and "top", define their meanings first. Otherwise, use the terms "base" and "limit", which is crystal clear.